lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php

```diff
@@ -157,14 +158,12 @@ public function setInflector(Inflector $inflector): void
     public function loadMetadataForClass(string $className, PersistenceClassMetadata $metadata): void
     {
         if (! $metadata instanceof ClassMetadata) {
-            Deprecation::trigger(
-                'doctrine/orm',
-                'https://github.com/doctrine/orm/pull/249',
-                'Passing an instance of %s to %s is deprecated, please pass a %s instance instead.',
-                $metadata::class,
+            throw new TypeError(sprintf(
+                'Argument #2 passed to %s() must be an instance of %s, %s given.',
                 __METHOD__,
                 ClassMetadata::class,
-            );
+                $metadata::class,
```

Comment:
```suggestion
                get_debug_type($metadata),
```

When embedding a class into a message, we should prefer `get_debug_type()` over `::class` because the former plays more nicely with anonymous classes.